Sources/Vapor/Middleware/FileMiddleware.swift

```diff
@@ -62,8 +62,9 @@ public final class FileMiddleware: Middleware {
             guard absPath.hasSuffix("/") else {
                 switch directoryAction.kind {
                 case .redirect:
+                    let redirectLocation = request.url.string.replacingOccurrences(of: request.url.path, with: "\(request.url.path)/")
                     return request.eventLoop.future(
-                        request.redirect(to: request.url.path + "/", redirectType: .permanent)
+                        request.redirect(to: redirectLocation, redirectType: .permanent)
```

Comment:
This is unsafe if the path occurs multiple times in the URL (which is perfectly possible). Try something like this, maybe:
```suggestion
                    var redirectUrl = request.url
                    redirectUrl.path += "/"
                    return request.eventLoop.future(request.redirect(to: redirectUrl.string, redirectType: .permanent))
```

Reply:
whoops, good catch. Updated.